src/overlays/actors/ovl_En_Door/z_en_door.h

```diff
@@ -5,30 +5,127 @@
+    /*  8 */ ENDOOR_SCH_TYPE_TREASURE_CHEST_SHOP,
+    /*  9 */ ENDOOR_SCH_TYPE_HONEY_DARLING_SHOP,
+    /* 10 */ ENDOOR_SCH_TYPE_MILK_BAR,
+    /* 11 */ ENDOOR_SCH_TYPE_INN_MAIN_ENTRANCE,
+    /* 12 */ ENDOOR_SCH_TYPE_INN_UPPER_ENTRANCE,
+    /* 13 */ ENDOOR_SCH_TYPE_INN_GRANNYS,
+    /* 14 */ ENDOOR_SCH_TYPE_INN_STAFF_ROOM,
+    /* 15 */ ENDOOR_SCH_TYPE_INN_KNIFE_CHAMBER,
+    /* 16 */ ENDOOR_SCH_TYPE_INN_LARGE_SUITE,
+    /* 17 */ ENDOOR_SCH_TYPE_MAYORS_RESIDENCE_MAIN_ENTRANCE,
+    /* 18 */ ENDOOR_SCH_TYPE_MAYORS_RESIDENCE_MAYOR_DOTOUR,
+    /* 19 */ ENDOOR_SCH_TYPE_MAYORS_RESIDENCE_MADAME_AROMA,
+    /* 20 */ ENDOOR_SCH_TYPE_MAYORS_RESIDENCE_BEDROOM,
+    /* 21 */ ENDOOR_SCH_TYPE_21, // unused
+    /* 22 */ ENDOOR_SCH_TYPE_ROMANI_RANCH_MAMAS_HOUSE,
+    /* 23 */ ENDOOR_SCH_TYPE_ROMANI_RANCH_BARN,
+    /* 24 */ ENDOOR_SCH_TYPE_ROMANI_RANCH_CUCCO_SHACK,
+    /* 25 */ ENDOOR_SCH_TYPE_ROMANI_RANCH_DOGGY_RACETRACK,
+    /* 26 */ ENDOOR_SCH_TYPE_ROMANI_RANCH_BEDROOM,
+    /* 27 */ ENDOOR_SCH_TYPE_IKANA_CANYON_MUSIC_BOX_HOUSE,
+    /* 28 */ ENDOOR_SCH_TYPE_DAMPES_HOUSE,
+    /* 29 */ ENDOOR_SCH_TYPE_MAGIC_HAGS_POTION_SHOP,
+    /* 30 */ ENDOOR_SCH_TYPE_30, // unused
+    /* 31 */ ENDOOR_SCH_TYPE_SWAMP_SHOOTING_GALLERY,
+    /* 32 */ ENDOOR_SCH_TYPE_MAX
+} EnDoorScheduleType;
+
+// ENDOOR_TYPE_FRAMED
+typedef enum EnDoorFramedType {
+    /* 0 */ ENDOOR_FRAMED_FRAME,
+    /* 1 */ ENDOOR_FRAMED_NOFRAME
+} EnDoorFramedType;
+
+#define ENDOOR_GET_TYPE(thisx) (((thisx)->params >> 7) & 7)
+#define ENDOOR_GET_ACTION_VAR(thisx) (((thisx)->params) & 0x7F)
+
+#define ENDOOR_GET_HALFDAYBIT_INDEX_FROM_HALFDAYCHECK(actionVar) ((actionVar) & 0x7)
+#define ENDOOR_GET_TEXTOFFSET_FROM_HALFDAYCHECK(actionVar) (((actionVar) >> 3) & 0xF)
+
+#define ENDOOR_PACK_ACTIONVAR_0_2_3(halfDayBitIndex, textOffset) ((((textOffset) & 0xF) << 3) | ((halfDayBitIndex) & 0x7))
+
+#define ENDOOR_PARAMS(type, actionVar) ((((type) & 7) << 7) | ((actionVar) & 0x7F))
+
+/**
+ * `scheduleType` must be a value of the `EnDoorScheduleType` enum
+ */
+#define ENDOOR_PARAMS_SCHEDULE(scheduleType) ENDOOR_PARAMS(ENDOOR_TYPE_SCHEDULE, scheduleType)
+
+/**
+ * `scheduleType` must be a value of the `EnDoorFramedType` enum
+ */
+#define ENDOOR_PARAMS_FRAMED(frameType) ENDOOR_PARAMS(ENDOOR_TYPE_FRAMED, frameType)
+
 typedef struct EnDoor {
     /* 0x000 */ KnobDoorActor knobDoor;
     /* 0x1A4 */ u8 doorType; // EnDoorType enum
-    /* 0x1A5 */ u8 switchFlag;
-    /* 0x1A6 */ u8 unk_1A6;
-    /* 0x1A7 */ s8 openTimer; // For how long the door will be open. positive/negative means the opening direction
+    /* 0x1A5 */ union {
+                    u8 actionVar; // generic
+                    u8 halfDayCheck; // ENDOOR_TYPE_WHOLE_DAY, ENDOOR_TYPE_DAY and ENDOOR_TYPE_NIGHT
+                    u8 switchFlag; // ENDOOR_TYPE_LOCKED
+                    u8 schType; // ENDOOR_TYPE_SCHEDULE
+                    u8 frameType; // ENDOOR_TYPE_FRAMED
+                } actionVar; // TODO: think on a better name
```

Comment:
I'm not completely sure if `actionVar` is a good name tbh. I'm open to suggestions

Reply:
Possibly name after type `typeVar` (still don't like `var` but can't think of a good way to remove it). would also probably say change the first element to something like just `data` that way it is generic and different then the union name.